### PR TITLE
WT-12413 Check in Workgen base/mirror tables are pending deletion at the same time

### PR DESCRIPTION
**Pull request by etienneptl**
We have a Testy error where a table does not seem the be found. Adding a check to make sure a base and its mirror are always being deleted together.

### CODE REVIEW

File: bench/workgen/workgen.cpp [hunk 1]
```diff
@@ -555,17 +555,25 @@ WorkloadRunner::select_table_for_drop(std::vector<std::string> &pending_delete)
...
     }
+
     schedule_table_for_drop(table_itr, pending_delete);
 
     // If the table has a mirror, schedule the mirror table for deletion too.
-    if (icontext->_dyn_table_runtime[table_itr->second].has_mirror()) {
-        auto mirror_itr =
-          icontext->_dyn_tint.find(icontext->_dyn_table_runtime[table_itr->second]._mirror);
+    if (has_mirror)
         schedule_table_for_drop(mirror_itr, pending_delete);
```
Comment (jiechenbo) [inline bench/workgen/workgen.cpp:576]: In the old state, if the table has a mirror table, we would call  `schedule_table_for_drop` which would assert already that the table must exist. Maybe we should add the `_pending_delete` assert in there?

File: bench/workgen/workgen.cpp [hunk 2]
```diff
@@ -555,17 +555,25 @@ WorkloadRunner::select_table_for_drop(std::vector<std::string> &pending_delete)
...
+    bool has_mirror = icontext->_dyn_table_runtime[table_itr->second].has_mirror();
+    auto mirror_itr = has_mirror ?
+      icontext->_dyn_tint.find(icontext->_dyn_table_runtime[table_itr->second]._mirror) :
+      icontext->_dyn_tint.end();
+
     if (icontext->_dyn_table_runtime[table_itr->second]._pending_delete) {
+        // The mirror should have been marked as pending deletion too.
+        if (has_mirror)
+            ASSERT(mirror_itr != icontext->_dyn_tint.end() &&
+              icontext->_dyn_table_runtime[mirror_itr->second]._pending_delete);
```
Comment (jiechenbo) [inline bench/workgen/workgen.cpp:568]: Maybe we should move to this assert, into when we are doing an table operation or before we commit a transaction.

Comment (etienneptl) [reply bench/workgen/workgen.cpp:568]: After checking the code, moving the check would be wrong. A table can be marked as pending deletion while threads are working on it even though they are not done with their transaction.
When marked for deletion, no new threads can use the table.

**Review by jiechenbo — APPROVED** [review]
lgtm!

**Comment by etienneptl** [comment]
Cancelling this PR as it was debugging code I added that did not end up being useful.

**Pull request closed without merging**